### PR TITLE
docs(Accordion): update API decision for modified Accordion

### PR DESCRIPTION
**Pull request by saurabhdaware**
## Changes in Accordion

- API Changed to add `AccordionItemHeader` and `AccordionItemBody` that allows having slot on Header as well
- The current API of `<AccordionItem title="" description="" />` will be deprecated and removed in next major version
- `variant` prop added that supports bordered and borderless variant in Accordion
- `size` prop added that supports medium-size accordions now

### CODE REVIEW

File: packages/blade/src/components/Accordion/_decisions/decisions.md [hunk 1]
```diff
@@ -30,6 +30,39 @@ Annotated component with props:
...
+    <AccordionItemBody>
+      <Text>
+        You can use Razorpay Route from the Dashboard or using APIs to transfer money to customers.
+      </Text>
+    </AccordionItemBody>
+  </AccordionItem>
+  {/* With custom slot for trigger */}
+  <AccordionItem>
+    <AccordionItemTrigger>
+      <div>Trigger Slot</div>
```
Comment (snitin315) [inline packages/blade/src/components/Accordion/_decisions/decisions.md:54]: In this case the props passed to `AccordionItemTrigger` will be ignored?

File: packages/blade/src/components/Accordion/_decisions/decisions.md [hunk 2]
```diff
@@ -63,12 +125,30 @@ import { Accordion, AccordionItem } from '@razorpay/blade';
...
+| variant                     | `bordered`, `borderless`        | `borderless` | Visual variant of AccordionItem                                                                                                                   |          |
+| children                    | `JSX`                           | `undefined`  | Accepts [`AccordionItemTrigger`](#accordionitemtrigger) and [`AccordionItemBody`](#accordionitembody). Accepting other JSX elements is deprecated | ✅       |
+| isExpanded                  | `boolean`                       | `undefined`  | Expanded state control when individual item is used                                                                                               |          |
+| onExpandChange              | (isExpanded: `boolean`) => void | `undefined`  | Callback for handling change in expand state when individual item is used                                                                         |          |
```
Comment (snitin315) [inline packages/blade/src/components/Accordion/_decisions/decisions.md:136]: Do we need these props? I think we should not allow individual usage of `AccordionItem`, it should always be wrapped within `Accordion`.

```js
<Accordion defaultExpandedIndex="0" onExpandChange={() => {}} >
 <AccordionItem>
    <AccordionItemTrigger
      title="Accordion Title"
      icon={QRIcon}
    />
    <AccordionItemBody>
      <Text>Information inside Accordion</Text>
    </AccordionItemBody>
  </AccordionItem>
 </Accordion>
```

File: packages/blade/src/components/Accordion/_decisions/decisions.md [hunk 3]
```diff
@@ -63,12 +125,30 @@ import { Accordion, AccordionItem } from '@razorpay/blade';
...
 | description | `string`        | `undefined` | Body text content                                                          |          |
 | icon        | `IconComponent` | `undefined` | Renders a Blade icon as title prefix (requires `showNumberPrefix={false}`) |          |
-| children    | `JSX`           | `undefined` | Slot, renders any custom content                                           |          |
+| children    | `JSX`           | `undefined` | Slot                                                                       |          |
+
+#### AccordionItemBody
+
+| Key      | Type  | Default     | Description | Required |
+| -------- | ----- | ----------- | ----------- | -------- |
+| children | `JSX` | `undefined` | Slot        | ✅       |
```
Comment (snitin315) [inline packages/blade/src/components/Accordion/_decisions/decisions.md:151]: Did you mean?

```suggestion
| children | `React.ReactNode` | `undefined` | Slot        | ✅       |
```

File: packages/blade/src/components/Accordion/_decisions/decisions.md [hunk 4]
```diff
@@ -117,16 +201,52 @@ const App = () => {
...
 ```
 
+#### With Individual Item
+
+```tsx
+const App = () => {
+  const [isOpen, setIsOpen] = useState(0);
+
+  return (
+    <AccordionItem
```
Comment (snitin315) [inline packages/blade/src/components/Accordion/_decisions/decisions.md:231]: I don't think we should allow this usage. It should always be wrapped within `Accordion` & the API remains consistent.

File: packages/blade/src/components/Accordion/_decisions/decisions.md [hunk 5]
```diff
@@ -150,9 +287,12 @@ Find in page (automatically expanding the accordion if someone does <kbd>Cmd</kb
 
 ## Open questions
 
-**Q1.** Is the deviation in props and using index for controlled state a fine tradeoff? Please post alternatives with usecase and rationale otherwise.
-
-**A1.** Yes.
+- **Q1.** Is the deviation in props and using index for controlled state a fine tradeoff? Please post alternatives with usecase and rationale otherwise.
+  - **Ans:** Yes.
+- **Q2.** `AccordionItemTrigger` vs `AccordionItemHeader` vs `AccordionItemControl`
```
Comment (snitin315) [inline packages/blade/src/components/Accordion/_decisions/decisions.md:292]: +1 for `AccordionHeader`.

File: packages/blade/src/components/Accordion/_decisions/decisions.md (hunk 2 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Accordion/_decisions/decisions.md:136]: yes I am yet to update it here

File: packages/blade/src/components/Accordion/_decisions/decisions.md (hunk 3 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Accordion/_decisions/decisions.md]: Yup

File: packages/blade/src/components/Accordion/_decisions/decisions.md (hunk 1 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Accordion/_decisions/decisions.md]: if you add `title`, it renders above the slot. This is configured in our BaseHeader

File: packages/blade/src/components/Accordion/_decisions/decisions.md [hunk 6]
```diff
@@ -150,9 +279,14 @@ Find in page (automatically expanding the accordion if someone does <kbd>Cmd</kb
...
-
-**A1.** Yes.
+- **Q1.** Is the deviation in props and using index for controlled state a fine tradeoff? Please post alternatives with usecase and rationale otherwise.
+  - **Ans:** Yes.
+- **Q2.** `AccordionItemHeader` vs `AccordionItemHeader` vs `AccordionItemControl`
+  - **Ans:** `AccordionItemHeader`
+- **Q3.** [Proposed API](#api) vs [Alternate API](#alternate-api-for-custom-trigger)
+  - **Ans:** Proposed API with deprecations for earlier API
+- **Q4.** `variant="bordered | borderless"` vs `variant="filled | transparent"`
+  - **Ans:** ⚠️ TBD
```
Comment (snitin315) [inline packages/blade/src/components/Accordion/_decisions/decisions.md:289]: I'd say `variant: "normal' | 'transparent"`

**Review by snitin315 — DISMISSED** [review]
LGTM

**Review by pcprashant — DISMISSED** [review]

File: packages/blade/src/components/Accordion/_decisions/decisions.md [hunk 7]
```diff
@@ -44,14 +77,45 @@ import { Accordion, AccordionItem } from '@razorpay/blade';
 </Accordion>;
 ```
 
+</details>
+
+#### Diff between deprecated version and new version
+
+1. AccordionItem accepts AccordionItemHeader and AccordionItemBody components now
+2. `title`, `icon` props from AccordionItem will be available on AccordionTrigger
```
Comment (kamleshchandnani) [inline packages/blade/src/components/Accordion/_decisions/decisions.md:85]:
```suggestion
2. `title`, `icon` props from AccordionItem will now be available on AccordionItemHeader
```

File: packages/blade/src/components/Accordion/_decisions/decisions.md [hunk 8]
```diff
@@ -63,12 +127,30 @@ import { Accordion, AccordionItem } from '@razorpay/blade';
...
+| icon (⚠️ deprecated)        | `IconComponent`   | `undefined` | Renders a Blade icon as title prefix (requires `showNumberPrefix={false}`)                                                                      |          |
+| children                    | `React.ReactNode` | `undefined` | Accepts [`AccordionItemHeader`](#AccordionItemHeader) and [`AccordionItemBody`](#accordionitembody). Accepting other JSX elements is deprecated | ✅       |
+
+#### AccordionItemHeader
+
+| Key         | Type              | Default     | Description                                              | Required |
+| ----------- | ----------------- | ----------- | -------------------------------------------------------- | -------- |
+| title       | `string`          | `undefined` | Title text content                                       |          |
+| subtitle    | `string`          | `undefined` | Subtitle text content                                    |          |
+| titleSuffix | `React.ReactNode` | `undefined` | Suffix part of title. Mostly for Badge                   |          |
```
Comment (kamleshchandnani) [inline packages/blade/src/components/Accordion/_decisions/decisions.md:143]: how will we prevent people accidentally using titleSuffix and trailing interchangeable?

Comment (saurabhdaware) [reply packages/blade/src/components/Accordion/_decisions/decisions.md:143]: It renders at 2 difference places. titleSuffix is immediately after title vs trailing is on the extreme right. So depending on the design they will use the correct parameter.

image

image

File: packages/blade/src/components/Accordion/_decisions/decisions.md [hunk 9]
```diff
@@ -30,6 +30,39 @@ Annotated component with props:
 
 Sample usage, composes `Accordion` and `AccordionItem`:
```
[the remark this answers is not in the document]
Comment (saurabhdaware) [reply packages/blade/src/components/Accordion/_decisions/decisions.md:28]: done

**Pull request merged**